pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -2258,54 +2258,73 @@ func updateRegistrationEntry(tx *gorm.DB,
-	if err := tx.Exec("DELETE FROM selectors WHERE registered_entry_id = ?", entry.ID).Error; err != nil {
-		return nil, sqlError.Wrap(err)
-	}
-
-	selectors := []Selector{}
-	for _, s := range req.Entry.Selectors {
-		selector := Selector{
-			Type:  s.Type,
-			Value: s.Value,
+	if req.Mask == nil || req.Mask.Selectors {
+		// Delete existing selectors - we will write new ones
+		if err := tx.Exec("DELETE FROM selectors WHERE registered_entry_id = ?", entry.ID).Error; err != nil {
+			return nil, sqlError.Wrap(err)
 		}
 
-		selectors = append(selectors, selector)
-	}
+		selectors := []Selector{}
+		for _, s := range req.Entry.Selectors {
+			selector := Selector{
+				Type:  s.Type,
+				Value: s.Value,
+			}
 
-	// Delete existing DNSs - we will write new ones
-	if err := tx.Exec("DELETE FROM dns_names WHERE registered_entry_id = ?", entry.ID).Error; err != nil {
-		return nil, sqlError.Wrap(err)
+			selectors = append(selectors, selector)
+		}
+		entry.Selectors = selectors
 	}
 
-	dnsList := []DNSName{}
-	for _, d := range req.Entry.DnsNames {
-		dns := DNSName{
-			Value: d,
+	if req.Mask == nil || req.Mask.DnsNames {
+		// Delete existing DNSs - we will write new ones
+		if err := tx.Exec("DELETE FROM dns_names WHERE registered_entry_id = ?", entry.ID).Error; err != nil {
+			return nil, sqlError.Wrap(err)
 		}
 
-		dnsList = append(dnsList, dns)
+		dnsList := []DNSName{}
+		for _, d := range req.Entry.DnsNames {
+			dns := DNSName{
+				Value: d,
+			}
+
+			dnsList = append(dnsList, dns)
+		}
+		entry.DNSList = dnsList
+	}
+
+	if req.Mask == nil || req.Mask.SpiffeId {
+		entry.SpiffeID = req.Entry.SpiffeId
+	}
+	if req.Mask == nil || req.Mask.ParentId {
+		entry.ParentID = req.Entry.ParentId
+	}
+	if req.Mask == nil || req.Mask.Ttl {
+		entry.TTL = req.Entry.Ttl
+	}
+	if req.Mask == nil || req.Mask.Admin {
+		entry.Admin = req.Entry.Admin
+	}
+	if req.Mask == nil || req.Mask.Downstream {
+		entry.Downstream = req.Entry.Downstream
+	}
+	if req.Mask == nil || req.Mask.EntryExpiry {
+		entry.Expiry = req.Entry.EntryExpiry
 	}
 
-	entry.SpiffeID = req.Entry.SpiffeId
-	entry.ParentID = req.Entry.ParentId
-	entry.TTL = req.Entry.Ttl
-	entry.Selectors = selectors
-	entry.Admin = req.Entry.Admin
-	entry.Downstream = req.Entry.Downstream
-	entry.Expiry = req.Entry.EntryExpiry
-	entry.DNSList = dnsList
 	if err := tx.Save(&entry).Error; err != nil {
 		return nil, sqlError.Wrap(err)
 	}
 
-	federatesWith, err := makeFederatesWith(tx, req.Entry.FederatesWith)
-	if err != nil {
-		return nil, err
-	}
+	if req.Mask == nil || req.Mask.FederatesWith {
+		federatesWith, err := makeFederatesWith(tx, req.Entry.FederatesWith)
+		if err != nil {
+			return nil, err
+		}
 
-	if err := tx.Model(&entry).Association("FederatesWith").Replace(federatesWith).Error; err != nil {
-		return nil, err
+		if err := tx.Model(&entry).Association("FederatesWith").Replace(federatesWith).Error; err != nil {
+			return nil, err
+		}
```

Comment:
We're supposed to return the updated entry, not req.Entry here. 
Oops, will fix.